tls/extensions/s2n_cookie.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/extensions/s2n_cookie.h"
+
+int s2n_cookie_len(struct s2n_connection *conn)
+{
+    return conn->cookie_data_len;
+}
+
+int s2n_cookie_recv(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    uint16_t cookie_len;
+
+    GUARD(s2n_stuffer_read_uint16(extension, &cookie_len));
+
+    if (cookie_len > S2N_MAX_COOKIE_SIZE) {
+        return 0;
+    }
```

Comment:
based on this we don't read cookies that their size larger than our selected random number. are we sure we want that?

Reply:
As per discussion yesterday, we want to be compatible with servers which may send larger cookies. So this has been refactored to use stuffers instead of a static array.